packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2591,6 +2608,10 @@ export class ContainerRuntime
 	 */
 	private processCore(messageWithContext: MessageWithContext) {
 		const { message, local } = messageWithContext;
+
+		// Intercept to reduce minimum sequence number to the delta manager's minimum sequence number.
+		messageWithContext.message.minimumSequenceNumber = this.deltaManager.minimumSequenceNumber;
```

Comment:
How do we ensure that we have enough validation in place that everything works fine?
At minimum, I'd assert here that MSN moves (with this change) only backwards. I.e. we could replace it with smaller number only on message.
Ideally, we would have an assert somewhere that it's noop if we have no pending messages or already connected and went through resubmit flow.